pkg/fanal/artifact/remote/git.go

```diff
@@ -48,9 +53,37 @@ func NewArtifact(rawurl string, c cache.ArtifactCache, artifactOpt artifact.Opti
 		cloneOptions.Progress = nil
 	}
 
-	_, err = git.PlainClone(tmpDir, false, &cloneOptions)
+	if len(artifactOpt.FetchCommit) == 0 {
+		cloneOptions.Depth = 1
+	}
+
+	if len(artifactOpt.FetchBranch) != 0 {
+		cloneOptions.ReferenceName = plumbing.ReferenceName("refs/heads/" + artifactOpt.FetchBranch)
```

Comment:
Why don't you use NewBranchReferenceName?

```suggestion
		cloneOptions.ReferenceName = plumbing.NewBranchReferenceName(artifactOpt.FetchBranch)
```

Reply:
Woah, did not notice this function at all. Thanks a lot, implementing this for tags as well as branches both!